server/src/resolvers/getVariantsResolver/utils/resolveChromosome.ts

```diff
@@ -0,0 +1,14 @@
+const resolveChromosome = (position: string) => {
+  const chromosome = position.match(/^[^:]*[^ :]/gm)?.join('');
+  const coordinates = position.match(/\w[^:]*$/gm)?.join('');
+  const end = coordinates?.match(/\w[^-]*$/gm);
+  const start = coordinates?.match(/^[^-]*[^ -]/gm);
```

Comment:
Were these expressions retrieved from elsewhere or did you write them yourself?

Reply:
The last two were written by me once I understand what the individual components like /, ^, [], * do.

The first two I was able to get by consulting this Stackoverflow here: https://stackoverflow.com/questions/3858519/regular-expression-to-match-everything-that-appears-before-a-colon

(Hopefully one day we will be able to write so much regex that we can pull them out of our brain without looking at anything...)

Reply:
Yeah, this is my main concern with regex, which is that it becomes incomprehensible even to the original writer some time after it was written. Often, they aren't even the right tool for the job because of their complexity and runtime (and risks of catastrophic backtracking). Thus I would recommend against them if the objective is achievable with more understandable code; for example, everything before a colon can also be obtained from `.split(":")`.

Reply:
That's a good point. I'll modify the code with split.